synapse/storage/databases/main/devices.py

```diff
@@ -1555,6 +1555,8 @@ async def add_device_change_to_streams(
 
         num_stream_ids = len(device_ids)
         if hosts:
+            # The `device_lists_outbound_pokes` wants a different stream ID for
+            # each row, which is a row per host per device update.
```

Comment:
```suggestion
            # `device_lists_outbound_pokes` wants a different stream ID for
            # each row, which is a row per host per device update.
            #
            # We use the same set of stream_ids in both `device_lists_stream`
            #  and `device_lists_outbound_pokes`, so we need a total of
            # max(len(device_ids), len(hosts) * len(device_ids)).
```

Reply:
oh well, apparently this is no longer correct.

Reply:
Ah yeah, it was surfacing a bug with the `get_all_device_list_changes_for_remotes` query. I thought this was how it worked previously, but that wasn't true so I've changed it to the old behaviour.